### PR TITLE
Updating variable names

### PR DESCRIPTION
**Pull request by rrtoledo**
## Content

Update variable names as discussed in https://github.com/cardano-scaling/alba/issues/38

## Comments

I changed `round_hash` (resp. `round_id`) to `hash` (resp. `id`) directly to dismiss clippy warnings.
As they are always used as `round.hash/id`, I thought this change was reasonable.

PR blocked by https://github.com/cardano-scaling/alba/pull/50
 
## Issue(s)

Relates to #38

### CODE REVIEW

File: src/centralized_telescope/setup.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+//! ALBA's Setup structure comprising, among others, the number of elements and internal parameters
+
+/// Setup output parameters
```
Comment (djetchev) [inline src/centralized_telescope/setup.rs:3]: Am I confused here, but I don't see the verifier threshold (the `n_f` in the paper)? Shouldn't it be part of the Setup struct?

File: src/centralized_telescope/setup.rs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#[derive(Debug, Clone, Copy)]
+pub struct Setup {
+    /// Approximate size of prover set to lower bound
```
Comment (djetchev) [inline src/centralized_telescope/setup.rs:6]: I'd rename this : `Total number of elements available to the prover.` The term `approximate size` may confuse the reader.

File: src/centralized_telescope/setup.rs (hunk 1 above)
Comment (rrtoledo) [reply src/centralized_telescope/setup.rs:3]: The verifier threshold (the n_f in the paper) is only needed for the setup parameter generations, hence it is not included in `Setup`

File: src/centralized_telescope/setup.rs (hunk 2 above)
Comment (rrtoledo) [reply src/centralized_telescope/setup.rs:6]: I have an issue with "total" (as `|Sp| >= np` only), but if you are mitigating it with "available". I'll update it now.

File: src/centralized_telescope/setup.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    pub set_size: u64,
+    /// Number of prover set's elements
+    pub proof_size: u64,
+    /// Maximum number of retries to find a proof
+    pub max_retries: u64,
+    /// Maximum number of subtrees to search to find a proof
+    pub search_width: u64,
+    /// Probability that a leaf was correctly chosen
```
Comment (djetchev) [inline src/centralized_telescope/setup.rs:14]: Now that I am thinking more and more about this name, we might be able to improve it even more as I'd be a bit confused about the term `leaf probability` if I am a first time reader. 

What is q: it is the probability that a random tuple `(t, s_1, \dots, s)u)` is a valid proof according to `h_2`. Maybe this can be said explicitly in the description to avoid confusion. Alternatively, it is the hardness. It sounds a bit more similar to `valid_proof_probability`?

File: src/centralized_telescope/setup.rs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+//! ALBA's Setup structure comprising, among others, the number of elements and internal parameters
+
+/// Setup output parameters
+#[derive(Debug, Clone, Copy)]
+pub struct Setup {
+    /// Approximate size of prover set to lower bound
+    pub set_size: u64,
+    /// Number of prover set's elements
```
Comment (djetchev) [inline src/centralized_telescope/setup.rs:8]: I'd change the comment to `The number of elements in a proof` to avoid confusion with `the total number of elements available to the prover at the proof generation`

File: src/utils/sample.rs [hunk 5]
```diff
@@ -7,16 +7,16 @@ use std::cmp::min;
 pub(crate) fn sample_uniform(hash: &[u8], n: u64) -> Option<u64> {
     // Computes the integer reprensation of hash* modulo n when n is not a
     // power of two. *(up to 8 bytes, in little endian)
-    fn mod_non_power_of_2(hash: &[u8], n: u64) -> Option<u64> {
+    fn mod_noset_sizeower_of_2(hash: &[u8], n: u64) -> Option<u64> {
```
Comment (djetchev) [inline src/utils/sample.rs:10]: confused with the function renaming

File: src/utils/sample.rs [hunk 6]
```diff
@@ -7,16 +7,16 @@ use std::cmp::min;
...
         fn log_base2(x: u64) -> u64 {
             u64::from(
                 u64::BITS
                     .saturating_sub(x.leading_zeros())
                     .saturating_sub(1),
             )
         }
-        let epsilon_fail: u64 = 1 << 40; // TODO: update
-        let k = log_base2(n.saturating_mul(epsilon_fail));
+        let epsilolower_boundail: u64 = 1 << 40; // TODO: update
```
Comment (djetchev) [inline src/utils/sample.rs:18]: what do we mean by this name ?

File: src/utils/sample.rs [hunk 7]
```diff
@@ -7,16 +7,16 @@ use std::cmp::min;
...
+        let k = log_base2(n.saturating_mul(epsilolower_boundail));
         let k_prime: u64 = 1 << k;
```
Comment (djetchev) [inline src/utils/sample.rs:20]: A bit puzzled about the variable name `k_prime` here considering we mean 2^k?

File: src/utils/sample.rs [hunk 8]
```diff
@@ -38,22 +38,22 @@ pub(crate) fn sample_uniform(hash: &[u8], n: u64) -> Option<u64> {
...
 
-/// Takes as input a hash and probability $q$ and returns true with
+/// Takes as input a hash and probability q and returns true with
 /// probability q otherwise false according to a Bernoulli distribution
 /// (c.f. Appendix B, Alba paper).
 pub(crate) fn sample_bernoulli(hash: &[u8], q: f64) -> bool {
     // For error parameter ɛ̝, find an approximation x/y of q with (x,y) in N²
     // such that 0 < q - x/y <= ɛ̝
-    let epsilon_fail: u64 = 1 << 40; // TOOD: update
+    let epsilolower_boundail: u64 = 1 << 40; // TOOD: update
```
Comment (djetchev) [inline src/utils/sample.rs:51]: idem - variable naming

File: src/centralized_telescope/proof.rs [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+/// Alba proof
+#[derive(Debug, Clone)]
+pub struct Proof {
+    /// Numbers of retries done to find the proof
+    pub retry_counter: u64,
+    /// Index of the searched subtree to find the proof
+    pub search_counter: u64,
+    /// Sequence of elements from prover set
+    pub element_sequence: Vec<Element>,
```
Comment (djetchev) [inline src/centralized_telescope/proof.rs:13]: this vector apparently contains the proof. Should we give it a more descriptive name such as `proof_sequence`, `proof_elements` or `proof_content`?

File: src/centralized_telescope/init.rs [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+            // Case 3, Theorem 14, i.e.  set_size >= λ^3
+            param_high_case(params, proof_size_f64, completeness_param2)
+        } else {
+            // Case 2, Theorem 13, i.e. λ^3 > set_size > λ^2
+            param_mid_case(params, proof_size_f64, s1)
+        }
+    }
+}
+
+fn param_small_case(params: &Params, proof_size_f64: f64) -> Setup {
```
Comment (djetchev) [inline src/centralized_telescope/init.rs:39]: I'd provide a comment to explain what `small`, `mid` and `high` proof means? If possible, we can even try to give more descriptive names but at least an explanation in the code is important.

File: src/centralized_telescope/proof.rs (hunk 9 above)
Comment (rrtoledo) [reply src/centralized_telescope/proof.rs:13]: The proof not only comprises a sequence of elements of the prover set but also the counters, as such `proof_content` might be misleading.

Each of these parameters are within a proof, and will be accessed as `proof.search/retry_counter" and `proof.element_sequence`. Repeating "proof" in the variable name seems a bit redundant in that aspect.

The reason why @curiecrypt and I liked the word "sequence" is because of the implied notion of order that you wouldn't have in "set" or "list".

File: src/utils/sample.rs (hunk 6 above)
Comment (rrtoledo) [reply src/utils/sample.rs:18]: This is a oversight of mine when changing all `n_f` to `lower_bound`. This variable is `epsilon_fail`.

File: src/utils/sample.rs (hunk 5 above)
Comment (rrtoledo) [reply src/utils/sample.rs:10]: This is a oversight of mine when changing all `n_p` to `set_size`.

File: src/centralized_telescope/init.rs (hunk 10 above)
Comment (rrtoledo) [reply src/centralized_telescope/init.rs:39]: There should be comments in the code above in `make_setup` (c.f. written below). Do you think they are not enough?
```
    if s1 < 1.0 || s2 < 1.0 {
        // Small case, i.e. set_size <= λ^2
        param_small_case(params, proof_size_f64)
    } else {
        let completeness_param2 = params.completeness_param.min(s2);
        if proof_size_f64 < completeness_param2 {
            // Case 3, Theorem 14, i.e.  set_size >= λ^3
            param_high_case(params, proof_size_f64, completeness_param2)
        } else {
            // Case 2, Theorem 13, i.e. λ^3 > set_size > λ^2
            param_mid_case(params, proof_size_f64, s1)
        }
    }
```

File: src/utils/sample.rs (hunk 7 above)
Comment (rrtoledo) [reply src/utils/sample.rs]: The name `k'` is not valid in rust so it was written in plain english. I'll update it to `new_n` (as it is used as `n` in `mod_power_2`)

File: src/centralized_telescope/setup.rs (hunk 3 above)
Comment (curiecrypt) [reply src/centralized_telescope/setup.rs:14]: I think, giving an explicit description would solve the confusion. Since we have a dfs implementation, the name leaf probability seems appropriate but needs an explicit description of course.

File: src/utils/sample.rs (hunk 5 above)
Comment (curiecrypt) [reply src/utils/sample.rs]: +1

File: src/centralized_telescope/init.rs (hunk 10 above)
Comment (curiecrypt) [reply src/centralized_telescope/init.rs:39]: We would give descriptive documentation about the parameter cases. @rrtoledo let's open an issue for that

**Review by curiecrypt — DISMISSED** [review]
LGTM 👍 
We should open issues for the topics that need to be documented.

File: src/centralized_telescope/setup.rs (hunk 3 above)
Comment (rrtoledo) [reply src/centralized_telescope/setup.rs:14]: We chose `leaf_probability` as we really had in mind the trees of tuples and named several variables and functions accordingly (`H1 -> node_hash`, `H2 -> leaf_hash`, `d -> search_width` for the search arity).
You can see the random tuple as a candidate proof that you have to check, we considered it as the lead linked to a round that we were checking.

I'd suggest that we either add more descriptive comments here to explain succinctly what we are doing with each structure or change all the tree related names for consistency (`H1 -> round_hash`, `H2 -> proof_hash`, `d -> ?`).

File: src/centralized_telescope/init.rs (hunk 10 above)
Comment (rrtoledo) [reply src/centralized_telescope/init.rs:39]: @curiecrypt  If it is just adding a few comments, we can do it here no? I have already added quite a few in `prove_index` and so on afterall.

**Review by djetchev — DISMISSED** [review]
Good work on this PR! 

LGTM

File: src/utils/sample.rs (hunk 7 above)
Comment (rrtoledo) [reply src/utils/sample.rs]: That was updated to `new_n`

File: src/centralized_telescope/setup.rs (hunk 3 above)
Comment (rrtoledo) [reply src/centralized_telescope/setup.rs:14]: Updated
